pkg/machine/shim/host.go

```diff
@@ -196,30 +196,33 @@ func Init(opts machineDefine.InitOptions, mp vmconfigs.VMProvider) error {
 	// copy it into the conf dir
 	if len(opts.IgnitionPath) > 0 {
 		err = ignBuilder.BuildWithIgnitionFile(opts.IgnitionPath)
-		return err
-	}
 
-	err = ignBuilder.GenerateIgnitionConfig()
-	if err != nil {
-		return err
-	}
+		if err != nil {
+			return err
+		}
+	} else {
+		err = ignBuilder.GenerateIgnitionConfig()
+		if err != nil {
+			return err
+		}
 
-	readyIgnOpts, err := mp.PrepareIgnition(mc, &ignBuilder)
-	if err != nil {
-		return err
-	}
+		readyIgnOpts, err := mp.PrepareIgnition(mc, &ignBuilder)
+		if err != nil {
+			return err
+		}
 
-	readyUnitFile, err := ignition.CreateReadyUnitFile(mp.VMType(), readyIgnOpts)
-	if err != nil {
-		return err
-	}
+		readyUnitFile, err := ignition.CreateReadyUnitFile(mp.VMType(), readyIgnOpts)
+		if err != nil {
+			return err
+		}
 
-	readyUnit := ignition.Unit{
-		Enabled:  ignition.BoolToPtr(true),
-		Name:     "ready.service",
-		Contents: ignition.StrToPtr(readyUnitFile),
+		readyUnit := ignition.Unit{
+			Enabled:  ignition.BoolToPtr(true),
+			Name:     "ready.service",
+			Contents: ignition.StrToPtr(readyUnitFile),
+		}
+		ignBuilder.WithUnit(readyUnit)
 	}
-	ignBuilder.WithUnit(readyUnit)
 
```

Comment:
The Hyper-V test is passing with this change
 
```suggestion
		err = ignBuilder.BuildWithIgnitionFile(opts.IgnitionPath)
		if err != nil {
			return err
		}

	} else {
		err = ignBuilder.GenerateIgnitionConfig()
		if err != nil {
			return err
		}
	}

	readyIgnOpts, err := mp.PrepareIgnition(mc, &ignBuilder)
	if err != nil {
		return err
	}

	readyUnitFile, err := ignition.CreateReadyUnitFile(mp.VMType(), readyIgnOpts)
	if err != nil {
		return err
	}

	readyUnit := ignition.Unit{
		Enabled:  ignition.BoolToPtr(true),
		Name:     "ready.service",
		Contents: ignition.StrToPtr(readyUnitFile),
	}
	ignBuilder.WithUnit(readyUnit)

```